oauthenticator/gitlab.py

```diff
@@ -162,6 +179,27 @@ def _check_group_whitelist(self, username, user_id, is_admin, access_token):
         return False
 
 
+    @gen.coroutine
+    def _check_project_id_whitelist(self, user_id, access_token):
+        http_client = AsyncHTTPClient()
```

Comment:
As you guessed it, complexity of batching is not necessary in this case since these calls are made only once per user at login. And very rarely would both group and project id would be specified. Most users would use one or the other from what I can tell.